### ISSUE TITLE
1 Table 2 nested sets

### ISSUE THREAD

**Issue by wdarins** [lazychaser/laravel-nestedset#442]
I need to have in same table 2 nested sets. So let's say for example I have:
```
class CategoryType1
{ 
  use NodeTrait;

  protected $table = 'categories';

  $fillable = [
    'name',
    'parent_id',
    '_lft',
    '_rgt',
    '_depth'
];
}
```
and 
```
class CategoryType2
{ 
  use NodeTrait;

  protected $table = 'categories';

  $fillable = [
      'name',
      'second_parent_id',
      '_s_lft',
      '_s_rgt',
      '_s_depth'
  ];

    public function getLftName(){
        return '_s_lft';
    }

    public function getRgtName(){
        return '_s_rgt';
    }

    public function getParentIdName(){
        return 'second_parent_id';
    }
}
```

And when I create a new node for parent like this:
```
$category= $parent->children()->create([
            'name' => 'TEST',
            '_binary_depth' => ($parent->_depth + 1)
        ]);
```

That creates a new record for `CategoryType1` but fills `CategoryType2` left and right with nulls. How would I set left and right the correct way without calling `CategoryType2::fixTree()` ?

Tried:

```
        $child = CategoryType2::find($category->id);
        $secondParent = CategoryType2::find($parent->id);
        $secondParent->appendNode($child);
        $child->update([
            '_s_depth' => $secondParent->_s_depth + 1
        ]);
```
That throws sql exception:
`SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'and  then `_s_lft`+2088 when `_s_lft` between  and 2088 then `_s_lft`-1 else ' at line 1`

Thanks.

**Comment by bjoernffm** [comment]
@wdarins , why do you want to save two nested sets in one table? If they're completely independent, they should be saved in seperate nested sets. Do you want to have different types of items in your nested set?

**Comment by wdarins** [comment]
@bjoernffm yes, I want to have 2 nested sets in same table. Let's say table users. When I query for nested set of type 1 I want to get the same users. Why would I duplicate the users table. That would make it hard for me to keep the data.

**Comment by wdarins** [comment]
I will give a concrete example: A user has a parent who maintains the chain. But at the same time this user has a sponsor. Think as a referal system.

**Comment by bjoernffm** [comment]
@wdarins I think I kinda got the idea, but you should consider a NoSQL solution. It sounds like traditional graph databases like Neo4j or Blazegraph could do the job for you :)

**Comment by wdarins** [comment]
@bjoernffm Thanks. But can't that be done with nested set? Just create a new model for the same table with other fields for left/right and parent id? And at user registration first create a simple User model and save and after that find that user with second model and make it a child? Is that a bad practice?

**Comment by wdarins** [comment]
The problem is that NodeTrait@moveNode does not update left and right attributes for existing node...

**Comment by wdarins** [comment]
@lazychaser Помоги плиз идеей

**Comment by wdarins** [comment]
I've solved it like this:
```
$child = CategoryType2::find($category->id);
$secondParent = CategoryType2::find($parent->id);

$position = $secondParent->_s_rgt;
$secondParent->newScopedQuery()->makeGap($position, 2);
$height = $sps->_s_rgt - $sps->_s_lft + 1;

$child->update([
    'second_parent_id' => $secondParent->id,
    '_s_depth' => $secondParent->_s_depth + 1,
    '_s_lft' => $position,
    '_s_rgt' => ($position + $height - 1),
]);
```
Even if that return that isBroken() true, descendants and other methods seems to work as expected. Still waiting to @lazychaser maybe a better answer. 
Thanks.
